lms/static/sass/shared/_course_object.scss

```diff
@@ -7,33 +7,44 @@
       width: flex-grid(4);
       @include margin-right(flex-gutter());
       float: left;
-
-      &:nth-child(3n+3) {
-        @include margin-right(0);
-      }
     }
 
     .courses-listing {
+      @include outer-container;
       @include clearfix();
       margin: 0;
       padding: 0;
       list-style: none;
 
       .courses-listing-item {
-        width: flex-grid(4);
-        @include margin-right(flex-gutter());
-        @include float(left);
+        height: 360px;
+
+        @include media($mobile) {
```

Comment:
@marjev, thanks for the reality check. My apologies, I misspoke there. Here are more thoughts:

* I'm with you on having Neat/Sass do the math for the width an element should take up given the ``$span`` value and the grid width for the current browser size/viewport.
* What was trying to get across in my previous (somewhat off-base) comment was that anything that is shared across viewports or needed by the default mobile/smallest viewport should be included outside of any media-query-based rules.
* If an element's width is the full width of the grid, I'd prefer we use the ``fill-parent()`` utility (http://thoughtbot.github.io/neat-docs/latest/#fill-parent) rather than defining the max number of columns of the grid at that breakpoint (since those column configurations can change over time).

With that said, my previous code example would look something like this:

```scss
.courses-listing-item {
    @include fill-parent();
    height: 360px; // please fix with above comments
    margin: 15px 0px;

    @include media($tablet) {
        @include span-columns(4); // 4 of 8
        ...
    }

    @include media($desktop) {
        @include span-columns(4); // 4 of 12
        ...
    }

    @include media($xl-desktop) {
        @include span-columns(3); // 3 of 12
        ...
    }
}
```

Hope that helps! Let me know if there's anything else confusing or just plain wrong about this.